UndertaleModTool/MainWindow.xaml.cs

```diff
@@ -772,58 +772,59 @@ private async void DataWindow_Closing(object sender, CancelEventArgs e)
                 e.Cancel = true;
 
                 bool save = false;
+                bool quit = false;
 
                 if (SettingsWindow.WarnOnClose)
                 {
-                    if (this.ShowQuestion("Are you sure you want to quit?") == MessageBoxResult.Yes)
-                    {
-                        if (this.ShowQuestion("Save changes first?") == MessageBoxResult.Yes)
-                        {
-                            if (scriptDialog is not null)
-                            {
-                                if (this.ShowQuestion("Script still runs. Save anyway?\nIt can corrupt the data file that you'll save.") == MessageBoxResult.Yes)
-                                    save = true;
-                            }
-                            else
-                                save = true;
+                    MessageBoxResult result = this.ShowQuestionWithCancel("Save changes before quitting?");
 
-                            if (save)
-                            {
-                                SaveResult saveRes = await SaveCodeChanges();
+                    if (result == MessageBoxResult.Yes) {
+                        quit = true;
 
-                                if (saveRes == SaveResult.NotSaved)
-                                    _ = DoSaveDialog();
-                                else if (saveRes == SaveResult.Error)
-                                {
-                                    this.ShowError("The changes in code editor weren't saved due to some error in \"SaveCodeChanges()\".");
-                                    return;
-                                }
-                            }
+                        if (scriptDialog is not null)
+                        {
+                            if (this.ShowQuestion("Script still runs. Save anyway?\nIt can corrupt the data file that you'll save.") == MessageBoxResult.Yes)
```

Comment:
please combine these two checks into one.

Reply:
I don't think they can be combined since if you answer no it should not set the save variable

Reply:
the can. `if (scriptDialog is not null && this.ShowQuestion(blabla)) save = true;`

Reply:
I don't think that works but now I realized this could work:

`if (scriptDialog is null || this.ShowQuestion(blabla)) save = true;`

Then remove the else.

Honestly though, who cares about such a small thing lol

Reply:
I feel that it improves readability much more, so would appreciate that being implemented.

Reply:
I think it's worse readability but sure thing boss